test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_neuron_hosting.py

```diff
@@ -62,11 +62,13 @@ def test_neuron_hosting(framework_version, ecr_image, instance_type, sagemaker_r
 @pytest.mark.model("resnet")
 @pytest.mark.processor("neuronx")
 @pytest.mark.parametrize(
-    "instance_type,sagemaker_regions",
+    "instance_type, sagemaker_regions",
     [("ml.trn1.2xlarge", ["us-east-1"]), ("ml.inf2.xlarge", ["us-east-2"])],
 )
 @pytest.mark.neuronx_test
 def test_neuronx_hosting(framework_version, ecr_image, instance_type, sagemaker_regions):
+    if "neuronx" not in ecr_image:
+        pytest.skip("Not a NeuronX DLC")
```

Comment:
We already have the autoused fixture `skip_by_device_type` which determines whether the test should be skipped if the instance type and test type do not match. Why can't this check be combined with that fixture?

Reply:
because skip_by_device_type does have the image variable in it and the "instance_type" is being parameterized on those 2 tests. this resolves the parameterized instance_type which is negating the "skip_by_device_type" without adding additional passes of variables, till TRN2 instances are available in PDX.